pkg/blob/nodeserver.go

```diff
@@ -339,21 +339,45 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		klog.V(3).Infof("edgecache attrib %v", attrib)
 		pvName, exists := attrib[pvNameKey]
 		var pv *v1.PersistentVolume
+		var err error
 		if exists {
-			pv, err = finalizer.GetPVByName(d.cloud.KubeClient, pvName)
+			pv, err = cache_volume.GetPVByName(d.cloud.KubeClient, pvName)
 		} else {
-			pv, err = finalizer.GetPVByVolumeID(d.cloud.KubeClient, volumeID)
+			pv, err = cache_volume.GetPVByVolumeID(d.cloud.KubeClient, volumeID)
 		}
 		if err != nil {
 			return nil, err
 		}
-		err = finalizer.AddFinalizer(d.cloud.KubeClient, pv, accountName, containerName)
-		if err != nil {
-			return nil, err
+
+		// attempt to figure out the name of the kube secret for the storage account key
+		if keyName == "" { // if the keyName wasn't already figured out by 'GetAuthEnv'
+			keyName, exists = attrib["secretName"]
+			if !exists { // if the keyName doesn't exist in the volume source context
+				keyName, exists = pv.ObjectMeta.Annotations[provisionerSecretNameField]
+				if !exists { // if keyName doesn't exist in the PV annotations
+					klog.Errorf("Failed to discover storage account key name.")
+					return nil, fmt.Errorf("Failed to discover storage account key name.")
+				}
+			}
 		}
-		if err = d.edgeCacheManager.EnsureVolume(accountName, accountKey, containerName, targetPath); err != nil {
+
+		// Pass this to RetryUpdatePVC to confidently add these annotations
+		var addAnnotations = func(inpvc *v1.PersistentVolumeClaim) *v1.PersistentVolumeClaim {
```

Comment:
might make sense to move this to edgecache manager, we have some other 'update pv'ish things there.